### ISSUE TITLE
Training Issues on Modified Code for EmbodiedOcc-Scannet-Mini Dataset

### ISSUE THREAD

**Issue by Made-Gpt** [YkiWu/EmbodiedOcc#5]
Thank you for open-sourcing this project!

I modified the code and trained the model on my device, but I encountered some issues with the results. Below are the last few lines of the training log:

``` 2025/01/20 22:53:29 - mmengine - INFO - [TRAIN] scene0332_01
2025/01/20 22:53:29 - mmengine - INFO - [TRAIN] Epoch 20 Iter    16/21   FocalLoss: 0.022 (0.022),   LovaszLoss: 0.833 (0.846),   Sem_Scal_Loss: 4.650 (4.373),   Geo_Scal_Loss: 2.089 (2.248),   Loss: 9.815 (9.645),   GradNorm: 11.422,   lr: 0.0000010,   time: 36.409 (0.224)
2025/01/20 22:54:04 - mmengine - INFO - [TRAIN] scene0692_03
2025/01/20 22:54:04 - mmengine - INFO - [TRAIN] Epoch 20 Iter    17/21   FocalLoss: 0.024 (0.018),   LovaszLoss: 0.801 (0.848),   Sem_Scal_Loss: 5.315 (4.875),   Geo_Scal_Loss: 2.198 (2.694),   Loss: 10.704 (10.231),   GradNorm: 8.847,   lr: 0.0000010,   time: 35.604 (0.229)
2025/01/20 22:54:39 - mmengine - INFO - [TRAIN] scene0258_00
2025/01/20 22:54:39 - mmengine - INFO - [TRAIN] Epoch 20 Iter    18/21   FocalLoss: 0.011 (0.022),   LovaszLoss: 0.747 (0.795),   Sem_Scal_Loss: 4.222 (3.394),   Geo_Scal_Loss: 2.011 (2.396),   Loss: 8.106 (8.771),   GradNorm: 7.328,   lr: 0.0000010,   time: 34.092 (0.209)
2025/01/20 22:55:15 - mmengine - INFO - [TRAIN] scene0168_01
2025/01/20 22:55:15 - mmengine - INFO - [TRAIN] Epoch 20 Iter    19/21   FocalLoss: 0.015 (0.015),   LovaszLoss: 0.863 (0.836),   Sem_Scal_Loss: 6.068 (5.291),   Geo_Scal_Loss: 2.435 (2.660),   Loss: 10.868 (10.265),   GradNorm: 9.032,   lr: 0.0000010,   time: 36.344 (0.228)
2025/01/20 22:55:52 - mmengine - INFO - [TRAIN] scene0000_00
2025/01/20 22:55:52 - mmengine - INFO - [TRAIN] Epoch 20 Iter    20/21   FocalLoss: 0.019 (0.018),   LovaszLoss: 0.831 (0.854),   Sem_Scal_Loss: 4.595 (4.746),   Geo_Scal_Loss: 2.514 (2.242),   Loss: 9.880 (9.640),   GradNorm: 8.402,   lr: 0.0000010,   time: 36.795 (0.274)
2025/01/20 22:55:53 - mmengine - INFO - Current global iou of sem is [ 0.779  0.000  0.178  0.015  0.032  0.066  0.000  0.054  0.065  0.000
  0.056  0.078]
2025/01/20 22:55:53 - mmengine - INFO - Current global iou of sem is 0.049394249105139745
2025/01/20 22:55:53 - mmengine - INFO - Current global iou of geo is 0.08473606774403745
2025/01/20 22:56:33 - mmengine - INFO - [EVAL] scene0089_00
2025/01/20 22:56:33 - mmengine - INFO - [EVAL] Iter     0/6   FocalLoss: 0.036 (0.021),   LovaszLoss: 0.841 (0.822),   Sem_Scal_Loss: 4.417 (4.237),   Geo_Scal_Loss: 2.527 (2.245),   Loss: 11.347 (9.431),   
2025/01/20 22:56:39 - mmengine - INFO - [EVAL] scene0593_01
2025/01/20 22:56:39 - mmengine - INFO - [EVAL] Iter     1/6   FocalLoss: 0.015 (0.020),   LovaszLoss: 0.859 (0.849),   Sem_Scal_Loss: 6.578 (5.200),   Geo_Scal_Loss: 3.154 (2.654),   Loss: 12.131 (10.687),   
2025/01/20 22:56:44 - mmengine - INFO - [EVAL] scene0272_01
2025/01/20 22:56:44 - mmengine - INFO - [EVAL] Iter     2/6   FocalLoss: 0.025 (0.021),   LovaszLoss: 0.851 (0.846),   Sem_Scal_Loss: 5.002 (5.120),   Geo_Scal_Loss: 2.088 (2.500),   Loss: 10.449 (10.610),   
2025/01/20 22:56:49 - mmengine - INFO - [EVAL] scene0416_01
2025/01/20 22:56:49 - mmengine - INFO - [EVAL] Iter     3/6   FocalLoss: 0.019 (0.022),   LovaszLoss: 0.867 (0.858),   Sem_Scal_Loss: 6.227 (5.495),   Geo_Scal_Loss: 2.827 (2.610),   Loss: 11.796 (11.155),   
2025/01/20 22:56:55 - mmengine - INFO - [EVAL] scene0510_02
2025/01/20 22:56:55 - mmengine - INFO - [EVAL] Iter     4/6   FocalLoss: 0.019 (0.021),   LovaszLoss: 0.856 (0.854),   Sem_Scal_Loss: 3.443 (5.217),   Geo_Scal_Loss: 2.516 (2.571),   Loss: 8.694 (10.775),   
2025/01/20 22:57:00 - mmengine - INFO - [EVAL] scene0107_00
2025/01/20 22:57:00 - mmengine - INFO - [EVAL] Iter     5/6   FocalLoss: 0.031 (0.023),   LovaszLoss: 0.890 (0.856),   Sem_Scal_Loss: 6.342 (5.351),   Geo_Scal_Loss: 1.927 (2.483),   Loss: 12.229 (10.941),   
2025/01/20 22:57:01 - mmengine - INFO - Current global iou of sem is [ 0.759  0.000  0.128  0.009  0.034  0.080  0.000  0.000  0.045  0.000
  0.005  0.022]
2025/01/20 22:57:01 - mmengine - INFO - Current global iou of sem is 0.029329690171255817
2025/01/20 22:57:01 - mmengine - INFO - Current global iou of geo is 0.06263113448194806
2025/01/20 22:57:01 - mmengine - INFO - Current single val iou of sem_cls is [ 0.823  0.000  0.122  0.007  0.029  0.081  0.000  0.000  0.025  0.000
  0.006  0.024]
2025/01/20 22:57:01 - mmengine - INFO - Current single val iou of sem is 0.026866777446333957
2025/01/20 22:57:01 - mmengine - INFO - Current single val iou of geo is 0.06283036537146594
2025/01/20 22:57:01 - mmengine - INFO - Current fov val iou of sem_cls is [ 0.702  0.000  0.155  0.011  0.024  0.097  0.000  0.000  0.023  0.000
  0.006  0.020]
2025/01/20 22:57:01 - mmengine - INFO - Current fov val iou of sem is 0.030576270754769062
2025/01/20 22:57:01 - mmengine - INFO - Current fov val iou of geo is 0.08128765580695951 
```

I trained on EmbodiedOcc-scannet-**mini** dataset, on **3** nvidia **3090**s. Dataset structure is prepared following readme.

**Modifications I Made**:
1. Data Preparation Code:
I added the following line to resolve an error regarding missing file paths:
```
local_img_paths = [path.replace('/data1/code/wyq/gaussianindoor/indoor-gaussian-scannet/data',  
                                self.ROOT) for path in valid_img_paths]  
```
2. Checkpoints:
I disabled the cfg.load_from section because no pretrained checkpoints were provided.

3. Absolute Paths:
I replaced absolute paths with relative paths in sys.path.join() to avoid potential issues.

**Questions and Concerns**:
1. Are these results expected when training from scratch on the EmbodiedOcc-mini dataset?
2. Could these low IoU values be related to the modifications I made, or is it more likely a configuration issue?
3. Are there any additional steps or specific configurations recommended for this dataset?

Thank you for your time and assistance! Let me know if you need more information or additional logs.

**Comment by YkiWu** [comment]
> Thank you for open-sourcing this project!
> 
> I modified the code and trained the model on my device, but I encountered some issues with the results. Below are the last few lines of the training log:
> 
> ```
> 2025/01/20 22:53:29 - mmengine - INFO - [TRAIN] Epoch 20 Iter    16/21   FocalLoss: 0.022 (0.022),   LovaszLoss: 0.833 (0.846),   Sem_Scal_Loss: 4.650 (4.373),   Geo_Scal_Loss: 2.089 (2.248),   Loss: 9.815 (9.645),   GradNorm: 11.422,   lr: 0.0000010,   time: 36.409 (0.224)
> 2025/01/20 22:54:04 - mmengine - INFO - [TRAIN] scene0692_03
> 2025/01/20 22:54:04 - mmengine - INFO - [TRAIN] Epoch 20 Iter    17/21   FocalLoss: 0.024 (0.018),   LovaszLoss: 0.801 (0.848),   Sem_Scal_Loss: 5.315 (4.875),   Geo_Scal_Loss: 2.198 (2.694),   Loss: 10.704 (10.231),   GradNorm: 8.847,   lr: 0.0000010,   time: 35.604 (0.229)
> 2025/01/20 22:54:39 - mmengine - INFO - [TRAIN] scene0258_00
> 2025/01/20 22:54:39 - mmengine - INFO - [TRAIN] Epoch 20 Iter    18/21   FocalLoss: 0.011 (0.022),   LovaszLoss: 0.747 (0.795),   Sem_Scal_Loss: 4.222 (3.394),   Geo_Scal_Loss: 2.011 (2.396),   Loss: 8.106 (8.771),   GradNorm: 7.328,   lr: 0.0000010,   time: 34.092 (0.209)
> 2025/01/20 22:55:15 - mmengine - INFO - [TRAIN] scene0168_01
> 2025/01/20 22:55:15 - mmengine - INFO - [TRAIN] Epoch 20 Iter    19/21   FocalLoss: 0.015 (0.015),   LovaszLoss: 0.863 (0.836),   Sem_Scal_Loss: 6.068 (5.291),   Geo_Scal_Loss: 2.435 (2.660),   Loss: 10.868 (10.265),   GradNorm: 9.032,   lr: 0.0000010,   time: 36.344 (0.228)
> 2025/01/20 22:55:52 - mmengine - INFO - [TRAIN] scene0000_00
> 2025/01/20 22:55:52 - mmengine - INFO - [TRAIN] Epoch 20 Iter    20/21   FocalLoss: 0.019 (0.018),   LovaszLoss: 0.831 (0.854),   Sem_Scal_Loss: 4.595 (4.746),   Geo_Scal_Loss: 2.514 (2.242),   Loss: 9.880 (9.640),   GradNorm: 8.402,   lr: 0.0000010,   time: 36.795 (0.274)
> 2025/01/20 22:55:53 - mmengine - INFO - Current global iou of sem is [ 0.779  0.000  0.178  0.015  0.032  0.066  0.000  0.054  0.065  0.000
>   0.056  0.078]
> 2025/01/20 22:55:53 - mmengine - INFO - Current global iou of sem is 0.049394249105139745
> 2025/01/20 22:55:53 - mmengine - INFO - Current global iou of geo is 0.08473606774403745
> 2025/01/20 22:56:33 - mmengine - INFO - [EVAL] scene0089_00
> 2025/01/20 22:56:33 - mmengine - INFO - [EVAL] Iter     0/6   FocalLoss: 0.036 (0.021),   LovaszLoss: 0.841 (0.822),   Sem_Scal_Loss: 4.417 (4.237),   Geo_Scal_Loss: 2.527 (2.245),   Loss: 11.347 (9.431),   
> 2025/01/20 22:56:39 - mmengine - INFO - [EVAL] scene0593_01
> 2025/01/20 22:56:39 - mmengine - INFO - [EVAL] Iter     1/6   FocalLoss: 0.015 (0.020),   LovaszLoss: 0.859 (0.849),   Sem_Scal_Loss: 6.578 (5.200),   Geo_Scal_Loss: 3.154 (2.654),   Loss: 12.131 (10.687),   
> 2025/01/20 22:56:44 - mmengine - INFO - [EVAL] scene0272_01
> 2025/01/20 22:56:44 - mmengine - INFO - [EVAL] Iter     2/6   FocalLoss: 0.025 (0.021),   LovaszLoss: 0.851 (0.846),   Sem_Scal_Loss: 5.002 (5.120),   Geo_Scal_Loss: 2.088 (2.500),   Loss: 10.449 (10.610),   
> 2025/01/20 22:56:49 - mmengine - INFO - [EVAL] scene0416_01
> 2025/01/20 22:56:49 - mmengine - INFO - [EVAL] Iter     3/6   FocalLoss: 0.019 (0.022),   LovaszLoss: 0.867 (0.858),   Sem_Scal_Loss: 6.227 (5.495),   Geo_Scal_Loss: 2.827 (2.610),   Loss: 11.796 (11.155),   
> 2025/01/20 22:56:55 - mmengine - INFO - [EVAL] scene0510_02
> 2025/01/20 22:56:55 - mmengine - INFO - [EVAL] Iter     4/6   FocalLoss: 0.019 (0.021),   LovaszLoss: 0.856 (0.854),   Sem_Scal_Loss: 3.443 (5.217),   Geo_Scal_Loss: 2.516 (2.571),   Loss: 8.694 (10.775),   
> 2025/01/20 22:57:00 - mmengine - INFO - [EVAL] scene0107_00
> 2025/01/20 22:57:00 - mmengine - INFO - [EVAL] Iter     5/6   FocalLoss: 0.031 (0.023),   LovaszLoss: 0.890 (0.856),   Sem_Scal_Loss: 6.342 (5.351),   Geo_Scal_Loss: 1.927 (2.483),   Loss: 12.229 (10.941),   
> 2025/01/20 22:57:01 - mmengine - INFO - Current global iou of sem is [ 0.759  0.000  0.128  0.009  0.034  0.080  0.000  0.000  0.045  0.000
>   0.005  0.022]
> 2025/01/20 22:57:01 - mmengine - INFO - Current global iou of sem is 0.029329690171255817
> 2025/01/20 22:57:01 - mmengine - INFO - Current global iou of geo is 0.06263113448194806
> 2025/01/20 22:57:01 - mmengine - INFO - Current single val iou of sem_cls is [ 0.823  0.000  0.122  0.007  0.029  0.081  0.000  0.000  0.025  0.000
>   0.006  0.024]
> 2025/01/20 22:57:01 - mmengine - INFO - Current single val iou of sem is 0.026866777446333957
> 2025/01/20 22:57:01 - mmengine - INFO - Current single val iou of geo is 0.06283036537146594
> 2025/01/20 22:57:01 - mmengine - INFO - Current fov val iou of sem_cls is [ 0.702  0.000  0.155  0.011  0.024  0.097  0.000  0.000  0.023  0.000
>   0.006  0.020]
> 2025/01/20 22:57:01 - mmengine - INFO - Current fov val iou of sem is 0.030576270754769062
> 2025/01/20 22:57:01 - mmengine - INFO - Current fov val iou of geo is 0.08128765580695951 
> ```
> 
> I trained on EmbodiedOcc-scannet-**mini** dataset, on **3** nvidia **3090**s. Dataset structure is prepared following readme.
> 
> **Modifications I Made**:
> 
> 1. Data Preparation Code:
>    I added the following line to resolve an error regarding missing file paths:
> 
> ```
> local_img_paths = [path.replace('/data1/code/wyq/gaussianindoor/indoor-gaussian-scannet/data',  
>                                 self.ROOT) for path in valid_img_paths]  
> ```
> 
> 2. Checkpoints:
>    I disabled the cfg.load_from section because no pretrained checkpoints were provided.
> 3. Absolute Paths:
>    I replaced absolute paths with relative paths in sys.path.join() to avoid potential issues.
> 
> **Questions and Concerns**:
> 
> 1. Are these results expected when training from scratch on the EmbodiedOcc-mini dataset?
> 2. Could these low IoU values be related to the modifications I made, or is it more likely a configuration issue?
> 3. Are there any additional steps or specific configurations recommended for this dataset?
> 
> Thank you for your time and assistance! Let me know if you need more information or additional logs.

**Comment by YkiWu** [comment]
Sorry for the late reply.
For the results you provided, the first to stress is that the Embodied results of our EmbodiedOcc rely on a pretrained model in local settings. You need to train this local model on the Occ-Scannet-mini2 dataset first. 
Secondly, due to the limited amount of data, the test set of EmbodiedOcc-Scannet-mini consists of only 16 scenes. If you perform testing with multiple GPUs, the experimental results will only include scenes processed by the first GPU (usually just a few scenes). These few scenes may not encompass all categories, leading to zeros in the results for some categories. The results in the paper for the mini set are obtained by testing all 16 scenes on a single GPU and calculating the metrics. 
If you have any questions, feel free to ask again.

**Comment by Made-Gpt** [comment]
Hi, I tried to train on occscennet-mini and embodiedocc-mini both on one 3090, the performance did look better!
